Signal/src/ViewControllers/ThreadSettings/OWSConversationSettingsViewController.m

```diff
@@ -1200,6 +1257,36 @@ - (void)otherUsersProfileDidChange:(NSNotification *)notification
     }
 }
 
+#pragma mark - ColorPickerDelegate
+
+- (void)showColorPicker
+{
+    ColorPickerViewController *pickerController = [[ColorPickerViewController alloc] initWithThread:self.thread];
+    pickerController.delegate = self;
+    OWSNavigationController *modal = [[OWSNavigationController alloc] initWithRootViewController:pickerController];
+
+    [self presentViewController:modal animated:YES completion:nil];
+}
+
+- (void)colorPicker:(ColorPickerViewController *)colorPicker didPickColorName:(NSString *)colorName
+{
+    DDLogDebug(@"%@ in %s picked color: %@", self.logTag, __PRETTY_FUNCTION__, colorName);
+    [self.editingDatabaseConnection asyncReadWriteWithBlock:^(YapDatabaseReadWriteTransaction * _Nonnull transaction) {
```

Comment:
Personally, I would make this sync, not async.  We _want_ to block on completion.  If we don't, the user could navigate before the completion is performed. TIOLI

Reply:
Done.